tests/adapters/test_visa.py

```diff
@@ -29,15 +29,43 @@
 from pymeasure.adapters import VISAAdapter
 from pymeasure.instruments import Instrument
 
-pyvisa_sim_installed = bool(importlib.util.find_spec('pyvisa_sim'))
+# This uses a pyvisa-sim default instrument, we could also define our own.
+SIM_RESOURCE = 'ASRL2::INSTR'
+
+is_pyvisa_sim_installed = bool(importlib.util.find_spec('pyvisa_sim'))
+if not is_pyvisa_sim_installed:
+    pytest.skip('PyVISA tests require the pyvisa-sim library', allow_module_level=True)
+
+
+def make_visa_adapter(**kwargs):
+    return VISAAdapter(SIM_RESOURCE, visa_library='@sim', **kwargs)
 
 
 def test_visa_version():
     assert VISAAdapter.has_supported_version()
 
 
-@pytest.mark.skipif(not pyvisa_sim_installed, reason='pyvisa-sim required but not found.')
 def test_correct_visa_kwarg():
     """Confirm that the query_delay kwargs gets passed through to the VISA connection."""
-    instr = Instrument(adapter='ASRL1::INSTR', name='delayed', query_delay=0.5, visa_library='@sim')
+    instr = Instrument(adapter=SIM_RESOURCE, name='delayed', query_delay=0.5, visa_library='@sim')
     assert instr.adapter.connection.query_delay == approx(0.5)
+
+
+def test_visa_adapter():
+    adapter = make_visa_adapter()
+    assert repr(adapter) == f"<VISAAdapter(resource='{SIM_RESOURCE}')>"
+
+    assert adapter.ask("*IDN?") == "SCPI,MOCK,VERSION_1.0\n"
+
+    adapter.write("*IDN?")
+    assert adapter.read() == "SCPI,MOCK,VERSION_1.0\n"
+
+
+def test_visa_adapter_ask_values():
+    adapter = make_visa_adapter()
+    assert adapter.ask_values(":VOLT:IMM:AMPL?", separator=",") == [1.0]
+
+
+def test_visa_adapter_write_binary_values():
+    adapter = make_visa_adapter()
+    adapter.write_binary_values("OUTP", [1], datatype='B')
```

Comment:
I think this test also needs to assert something. I played around with what I think you wanted to do, but could not get it to work - does pyvisa-sim work correctly with `write_binary_values`?

Reply:
Good question. I was looking for a way to see what is written. We could use `mocker` from `pytest-mock` to ensure that the right call is being made. Ideally `pyvisa-sim` has a place for that already, but I haven't found it yet.

Reply:
I agree that we should improve this test, but am thinking to leave it as a future improvement. There is a lot we can do with `pyvisa-sim`, but I don't want to go too deep in this PR.